src/merkle/store/mod.rs

```diff
@@ -430,3 +431,45 @@ impl MerkleStore {
         }
     }
 }
+
+impl Serializable for Node {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        target.write_bytes(&self.left.as_bytes());
+        target.write_bytes(&self.right.as_bytes());
+    }
+}
+
+impl Deserializable for Node {
+    fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
+        let left = RpoDigest::read_from(source)?;
+        let right = RpoDigest::read_from(source)?;
+        Ok(Node { left, right })
+    }
+}
+
+impl Serializable for MerkleStore {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        let len = self.nodes.len() as u64;
+        target.write_bytes(&len.to_be_bytes());
```

Comment:
You should be able to replace this with:
```Rust
target.write_u64(self.nodes.len() as u64);
```

Reply:
done